hail/python/hailtop/batch_client/aioclient.py

```diff
@@ -358,19 +357,21 @@ def create_job(self, image, command, env=None, mount_docker_socket=False,
         batch = Batch(self._client, b['id'], b.get('attributes'))
 
         specs = []
+        runner = AsyncThrottledGather(2)
         n = 0
         for job_spec in self._job_specs:
             n += 1
             specs.append(job_spec)
             if n == job_array_size:
-                await self.pool.call(self._submit_job, batch.id, specs)
+                await runner.call(self._submit_job, batch.id, specs)
                 n = 0
                 specs = []
 
         if specs:
-            await self.pool.call(self._submit_job, batch.id, specs)
+            await runner.call(self._submit_job, batch.id, specs)
 
-        await self.pool.wait()
+        if len(self._job_specs) > 0:
+            await runner.wait()
```

Comment:
This interface feels like a footgun. It seems sensible to start the Event triggered and untrigger it whenever a job is submitted.